packages/components/src/languageSupport/squiggle.ts

```diff
@@ -4,86 +4,134 @@ import {
+              LambdaParameter: t.variableName,
+            }),
+            foldNodeProp.add({
+              LambdaExpr: (context) => ({
+                from: context.getChild("NonEmptyProgram")?.from || 0,
+                to: context.getChild("NonEmptyProgram")?.to || 0,
+              }),
+              BlockExpr: foldInside,
+              RecordExpr: foldInside,
+              ArrayExpr: foldInside,
+            }),
+            indentNodeProp.add({
+              RecordExpr: (context) =>
+                context.baseIndent +
+                (context.textAfter === "}" ? 0 : context.unit),
+              BlockExpr: (context) =>
+                context.baseIndent +
+                (context.textAfter === "}" ? 0 : context.unit),
+              LambdaExpr: (context) =>
+                context.baseIndent +
+                (context.textAfter === "}" ? 0 : context.unit),
+              ArrayExpr: (context) =>
+                context.baseIndent +
+                (context.textAfter === "]" ? 0 : context.unit),
+            }),
+          ],
         }),
-        BlockExpr: foldInside,
-        RecordExpr: foldInside,
-        ArrayExpr: foldInside,
-      }),
-      indentNodeProp.add({
-        RecordExpr: (context) =>
-          context.baseIndent + (context.textAfter == "}" ? 0 : context.unit),
-        BlockExpr: (context) =>
-          context.baseIndent + (context.textAfter == "}" ? 0 : context.unit),
-        LambdaExpr: (context) =>
-          context.baseIndent + (context.textAfter == "}" ? 0 : context.unit),
-        ArrayExpr: (context) =>
-          context.baseIndent + (context.textAfter == "]" ? 0 : context.unit),
-      }),
-    ],
-  }),
-  languageData: {
-    commentTokens: {
-      line: "#",
-    },
-    autocomplete: [
-      snippetCompletion("{ |${args}| ${body} ", {
-        label: "{",
-        detail: "lambda function",
-      }),
-    ],
-    closeBrackets: {
-      brackets: ['"', "'", "(", "{"],
-    },
-  },
-});
+        languageData: {
+          commentTokens: {
+            line: "#",
+          },
+          autocomplete: (cmpl: CompletionContext) => {
+            const tree = syntaxTree(cmpl.state);
+            const tag = tree.cursorAt(cmpl.pos);
+            const defaultCompletion = {
+              from: tag.from,
+              options: [
+                /*eslint no-template-curly-in-string: "off"*/
+                snippetCompletion("{ |${args}| ${body} ", {
+                  label: "{",
+                  detail: "lambda function",
+                  type: "syntax",
+                }),
+              ],
+            };
+
+            const field = cmpl.tokenBefore(["AccessIdentifier"]);
+            if (field === null) {
+              return defaultCompletion;
+            }
+            const from = field.from;
 
-function squiggle() {
-  return new LanguageSupport(squiggleLang, []);
+            const cursor = tree.cursor();
+            var names: String[] = [];
```

Comment:
Ugh, `var`. Should be `let` (or `const` is also fine).

Reply:
I kinda hate the fact that const only means the reference is immutable, and var really does look more descriptive of what's happening.

But okay, sure.

Reply:
Yep, I often use `let` on mutable arrays and objects for the same reason. But never `var`.